test/test_grid_search.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import fairlearn.grid_search as gs
+
+import numpy as np
+import pandas as pd
+
+import simple_learners
+
+import pytest
+
+class TestGridSearch:
+    def _quick_data(self, number_samples=8):
+        feature_1 = np.random.randint(2, size=number_samples)
```

Comment:
Could this result in tests passing randomly? I'm a little worried that we'll hit failures and then we won't be able to repro.

Reply:
I was just about to say 'no' but I've just hit a case where it does. If all the attributes are randomly chosen to be 0 or 1. I'm guessing that's another 'trivial' case which we should allow..

Reply:
There is another case you can see when I'm checking that '0' appears in the automatically generated set of Lagrange multipliers. The np.linspace method doesn't always land exactly on zero :-/

Reply:
Correct, so that kind of randomness is not necessarily desirable.

linspace just produces floating point numbers, so you'd probably have to put in some tolerance?

Reply:
Indeed - I should have known better. Even trying to rig things by using powers of two (hence making the floating point representation exact) didn't quite work

Reply:
Things should be reliable now